packages/cozy-harvest-lib/src/components/KonnectorConfiguration/ConfigurationTab/EditContract.jsx

```diff
@@ -0,0 +1,279 @@
+  const [owners, setOwners] = useState(getAccountOwners(account))
+
+  const handleSubmit = e => {
+    e.preventDefault()
+    onSubmit({ shortLabel, owners })
+  }
+
+  const fieldVariant = isMobile ? 'default' : 'inline'
+
+  return (
+    <form onSubmit={handleSubmit} {...rest}>
+      <Stack spacing={isMobile ? 's' : 'm'}>
+        <Field
+          id="account-label"
+          label={t('contractForm.label')}
+          value={shortLabel}
+          onChange={e => setShortLabel(e.target.value)}
+          variant={fieldVariant}
+        />
+        <CollectionField
+          label={t('contractForm.owner')}
+          values={owners}
+          component={ContactPicker}
+          addButtonLabel={t('contractForm.addOwnerBtn')}
+          removeButtonLabel={t('contractForm.removeOwnerBtn')}
+          variant={fieldVariant}
+          onChange={owners => setOwners(owners)}
+          placeholder={t('contractForm.ownerPlaceholder')}
+        />
+        <Field
+          id="account-institution"
+          label={t('contractForm.bank')}
+          value={getAccountInstitutionLabel(account)}
+          disabled
+          variant={fieldVariant}
+        />
+        <Field
+          id="account-number"
+          label={t('contractForm.number')}
+          value={account.number}
+          disabled
+          variant={fieldVariant}
+        />
+        <Field
+          id="account-type"
+          label={t('contractForm.type')}
+          value={getAccountType(account)}
+          disabled
+          variant={fieldVariant}
+        />
+      </Stack>
+      <FormControls onCancel={onCancel} t={t} />
+    </form>
+  )
+}
+
+const FormControls = props => {
+  const { t } = useI18n()
+  const { onCancel, form, ...rest } = props
+
+  return (
+    <div className="u-mt-2" {...rest}>
+      <Button
+        label={t('contractForm.cancel')}
+        theme="secondary"
+        onClick={onCancel}
+      />
+      <Button
+        type="submit"
+        form={form}
+        label={t('contractForm.apply')}
+        theme="primary"
+      />
+    </div>
+  )
+}
+
+const DeleteConfirm = ({
+  cancel,
+  confirm,
+  title,
+  description,
+  secondaryText,
+  primaryText
+}) => {
+  return (
+    <Modal
+      title={title}
+      description={<div dangerouslySetInnerHTML={{ __html: description }} />}
+      secondaryType="secondary"
+      secondaryText={secondaryText}
+      secondaryAction={cancel}
+      dismissAction={cancel}
+      primaryType="danger"
+      primaryText={primaryText}
+      primaryAction={confirm}
+    />
+  )
+}
+
+const saveAccount = async params => {
+  const { client, account, fields, onSuccess, onError } = params
+
+  Object.keys(fields).forEach(key => {
+    if (key === 'owners') {
+      set(
+        account,
+        'relationships.owners.data',
+        fields[key].map(owner => ({
+          _id: owner._id,
+          _type: owner._type
+        }))
+      )
```

Comment:
it's a bit opaque for someone not dealing with account on daily basis. Are you adding a `owners` relationship to an `io.cozy.account` ?

Reply:
You're right, I will make this clearer.